streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -781,4 +782,18 @@ public void shouldNotAllowToAddGlobalStoreWithSourceNameEqualsProcessorName() {
             sameNameForSourceAndProcessor,
             new MockProcessorSupplier());
     }
+
+    @Test
+    public void shouldCompareSourceNode() {
```

Comment:
Just a basic test for `null` pattern.

Reply:
nit: Could you rename this test so that it reflects the focus on the `null` pattern?

Reply:
Ack. Renamed this one and added a new test for Pattern case.

Note, we know that either one of both is `null`.